tmail-backend/tmail-third-party/openpaas/src/main/java/com/linagora/tmail/CalDavEventAttendanceRepository.java

```diff
@@ -0,0 +1,286 @@
+        }
+    }
+
+    private CalendarEventAttendanceResults getAttendanceStatusFromCalendarEvent(BlobId blobId, Username username, CalendarEventParsed event) throws AddressException {
+        return event.participants()
+            .findParticipantByMailTo(username.asMailAddress().toString())
+            .flatMap(CalendarAttendeeField::participationStatus)
+            .map(status -> CalendarEventAttendanceResults$.MODULE$.done(createEventAttendanceStatusEntry(blobId, status)))
+            .getOrElse(() -> CalendarEventAttendanceResults$.MODULE$.notFound(blobId));
+    }
+
+    private static EventAttendanceStatusEntry createEventAttendanceStatusEntry(BlobId blobId, CalendarAttendeeParticipationStatus attendanceStatus) {
+        return new EventAttendanceStatusEntry(blobIdToString(blobId),
+            AttendanceStatus.fromCalendarAttendeeParticipationStatus(attendanceStatus).orElseThrow());
+    }
+
+    private Mono<DavUser> toDavUser(Username username) {
+        try {
+            return davUserProvider.provide(username.asMailAddress())
+                .switchIfEmpty(Mono.error(new RuntimeException("Unable to find user in Dav server with username '%s'".formatted(username.asString()))));
+        } catch (AddressException e) {
+            LOGGER.debug("Failed to get user DAV information", e);
+            return Mono.empty();
+        }
+    }
+
+    @Override
+    public Publisher<CalendarEventReplyResults> setAttendanceStatus(Username username,
+                                                                    AttendanceStatus attendanceStatus,
+                                                                    BlobIds eventBlobIds,
+                                                                    Optional<LanguageLocation> maybePreferredLanguage) {
+        MailboxSession session = sessionProvider.createSystemSession(username);
+
+        return toDavUser(username)
+            .flatMapMany(davUser ->
+                blobIdsAsFlux(eventBlobIds)
+                    .flatMap(blobId -> updateAttendanceStatusForEventBlob(blobIdFromRefinedId(blobId), attendanceStatus, davUser, session)));
+    }
+
+    private Mono<CalendarEventReplyResults> updateAttendanceStatusForEventBlob(BlobId blobId, AttendanceStatus attendanceStatus, DavUser davUser, MailboxSession session) {
+        PartStat partStat = attendanceStatus.toPartStat().orElseThrow();
+        return extractMessageId(blobIdToString(blobId))
+            .flatMap(messageId ->
+                Mono.fromDirect(messageIdManager.getMessagesReactive(List.of(messageId), FetchGroup.HEADERS, session))
+                    .map(this::retrieveEventUid)
+                    .flatMap(eventUid -> doGetCalendarObjectContainingVEvent(davUser, eventUid))
+                    .flatMap(targetCalendarObject ->
+                        updateCalendarObject(davUser, targetCalendarObject.uri(), partStat))
```

Comment:
```suggestion
                    .flatMap(targetCalendarObject -> updateCalendarObject(davUser, targetCalendarObject.uri(), partStat))
```

:-(

Seriously this level of comments is a waste of my time